compiler_gym/envs/client_service_compiler_env.py

```diff
@@ -0,0 +1,1372 @@
+
+        In normal use you should use :code:`gym.make(...)` rather than calling
+        the constructor directly.
+
+        :param service: The hostname and port of a service that implements the
+            CompilerGym service interface, or the path of a binary file which
+            provides the CompilerGym service interface when executed. See
+            :doc:`/compiler_gym/service` for details.
+
+        :param rewards: The reward spaces that this environment supports.
+            Rewards are typically calculated based on observations generated by
+            the service. See :class:`Reward <compiler_gym.spaces.Reward>` for
+            details.
+
+        :param benchmark: The benchmark to use for this environment. Either a
+            URI string, or a :class:`Benchmark
+            <compiler_gym.datasets.Benchmark>` instance. If not provided, the
+            first benchmark as returned by
+            :code:`next(env.datasets.benchmarks())` will be used as the default.
+
+        :param observation_space: Compute and return observations at each
+            :func:`step()` from this space. Accepts a string name or an
+            :class:`ObservationSpaceSpec
+            <compiler_gym.views.ObservationSpaceSpec>`. If not provided,
+            :func:`step()` returns :code:`None` for the observation value. Can
+            be set later using :meth:`env.observation_space
+            <compiler_gym.envs.ClientServiceCompilerEnv.observation_space>`. For available
+            spaces, see :class:`env.observation.spaces
+            <compiler_gym.views.ObservationView>`.
+
+        :param reward_space: Compute and return reward at each :func:`step()`
+            from this space. Accepts a string name or a :class:`Reward
+            <compiler_gym.spaces.Reward>`. If not provided, :func:`step()`
+            returns :code:`None` for the reward value. Can be set later using
+            :meth:`env.reward_space
+            <compiler_gym.envs.ClientServiceCompilerEnv.reward_space>`. For available spaces,
+            see :class:`env.reward.spaces <compiler_gym.views.RewardView>`.
+
+        :param action_space: The name of the action space to use. If not
+            specified, the default action space for this compiler is used.
+
+        :param derived_observation_spaces: An optional list of arguments to be
+            passed to :meth:`env.observation.add_derived_space()
+            <compiler_gym.views.observation.Observation.add_derived_space>`.
+
+        :param connection_settings: The settings used to establish a connection
+            with the remote service.
+
+        :param service_connection: An existing compiler gym service connection
+            to use.
+
+        :raises FileNotFoundError: If service is a path to a file that is not
+            found.
+
+        :raises TimeoutError: If the compiler service fails to initialize within
+            the parameters provided in :code:`connection_settings`.
+        """
```

Comment:
Does this need to invoke gym.Env superclass constructor?

Reply:
Neither `CompilerEnv` or `gym.Env` have `__init__(...)`.